obspy/io/nordic/ellipse.py

```diff
@@ -0,0 +1,329 @@
+            center=x,y coordinates of ellipse center (0,0)
+        """
+        if a < b:
+            print('Semi-major smaller than semi-minor! Switching...')
+            self.a = b
+            self.b = a
+        else:
+            self.a = a
+            self.b = b
+        self.theta = theta
+        self.x = center[0]
+        self.y = center[1]
+
+    @classmethod
+    def from_cov(cls, cov, center=(0, 0), debug=False):
+        """Set error ellipse using covariance matrix
+
+        Call as e=_ellipse.from_cov(cov,center)
+
+        Inputs:
+            cov: covariance matrix [[c_xx, c_xy], [c_xy, c_yy]]
+            center:  center position (x,y)
+        Sources:
+            http://www.visiondummy.com/2014/04/
+                   draw-error-ellipse-representing-covariance-matrix/
+            https://blogs.sas.com/content/iml/2014/07/23/
+                    prediction-ellipses-from-covariance.html
+        """
+        evals, evecs = np.linalg.eig(cov)
+        evals = np.sqrt(evals)
+        if debug:
+            print(evecs)
+            print(evals)
+        # Sort eigenvalues in decreasing order
+        sort_indices = np.argsort(evals)[::-1]
+        # Select semi-major and semi-minor axes
+        a, b = evals[sort_indices[0]], evals[sort_indices[1]]
+        # Calculate angle of semi-major axis
+        x_v1, y_v1 = evecs[:, sort_indices[0]]
+        if y_v1 == 0.:
+            theta = 90.
+        else:
+            theta = (np.degrees(np.arctan((x_v1)/(y_v1))) + 180) % 180
+        return cls(a, b, theta, center)
+
+    @classmethod
+    def from_uncerts(cls, x_err, y_err, c_xy, center=(0, 0), debug=False):
+        """Set error ellipse using Nordic epicenter uncertainties
+
+        Call as e=_ellipse.from_uncerts(x_err,y_err,c_xy,center)
+
+        Inputs:
+            x_err: x error (m)
+            y_err: y error (m)
+            c_xy:  x-y covariance (m^2)
+            center:  center position (x,y)
+        """
+        cov = [[x_err**2, c_xy], [c_xy, y_err**2]]
+        return cls.from_cov(cov, center)
+
+    @classmethod
+    def from_uncerts_baz(cls, x_err, y_err, c_xy, dist, baz,
+                         viewpoint=(0, 0)):
+        """Set error ellipse using uncertainties, distance and back-azimuth
+
+        Call as e=_ellipse.from_uncerts_baz(xerr,yerr,c_xy,dist,baz[,viewpt])
+
+        Inputs:
+            x_err: x error (m)
+            y_err: y error (m)
+            c_xy:  x-y covariance (m^2)
+            dist:  distance of center from observer
+            baz:   back-azimuth from observer (degrees)
+            viewpoint: observer's position [(0,0)]
+        """
+        x = viewpoint[0] + dist*np.sin(np.radians(baz))
+        y = viewpoint[1] + dist*np.cos(np.radians(baz))
+        return cls.from_uncerts(x_err, y_err, c_xy, (x, y))
+
+    def to_cov(self, debug=False):
+        """Convert to covariance matrix notation
+
+        Call as cov, center =myellipse.to_uncerts()
```

Comment:
Looks like this doc-string needs correcting

Reply:
I think I corrected this with the other docstrings